src/worker/worker.ts

```diff
@@ -0,0 +1,37 @@
+import { readExcel } from "./read-wordlist";
+import { WorkerWrapper } from "@common/worker-wrapper";
+import {
+  compileModelFromLexicalModelSource,
+  WordListFromArray,
+} from "@predictive-text-studio/lexical-model-compiler";
+import { linkStorageToKmp } from "./link-storage-to-kmp";
+import Storage from "./storage";
+
+export class WorkerWrapperImpl implements WorkerWrapper {
+  private storage: Storage;
+
+  constructor(storage = new Storage()) {
+    this.storage = storage;
+  }
+
+  saveFile = async (name: string, file: File): Promise<ArrayBuffer> => {
+    const wordlist = await readExcel(await file.arrayBuffer());
+    this.storage.saveFile(name, wordlist);
+    return await linkStorageToKmp(this.storage);
+  };
+
+  compileModel = async (): Promise<string> => {
+    // TODO: Parse multiple dictionary sources, right now just reading the first file
+    const storedFiles = await this.storage.fetchAllFiles();
+    if (storedFiles.length < 1) {
+      throw new Error("Cannot find any file in the IndexedDB");
+    } else {
+      const file = storedFiles[0];
+      const code = compileModelFromLexicalModelSource({
+        format: "trie-1.0",
+        sources: [new WordListFromArray(file.name, file.wordlist)],
+      });
+      return code;
+    }
```

Comment:
After reading this, I realized that the two functions we export as part of the worker are _wacky_ and make no sense together, but, as I mentioned before, I think that's an issue for a different PR...